packages/protocol/scripts/truffle/upgrade.ts

```diff
@@ -36,14 +37,43 @@ function getProxiedContracts() {
     return proxyFilename.slice(0, -'Proxy.json'.length)
   }
 
-  return fs
-    .readdirSync(contractsDir)
-    .filter((filename: string) => /\w+Proxy.json$/.test(filename))
-    .map(contractNameFromProxyFilename)
-    .map((contractName: string) => [
+  let names = []
+  if (argv.contract) {
+    names.push(argv.contract)
+  } else {
+    names = fs
+      .readdirSync(contractsDir)
+      .filter((filename: string) => /\w+Proxy.json$/.test(filename))
+      .map(contractNameFromProxyFilename)
+  }
+
+  return names.map((contractName: string) => {
+    // tslint:disable-next-line:no-console
+    console.log('Reading artifact', contractName)
+    return [
       artifacts.require(contractName) as Truffle.Contract<any>,
       artifacts.require(contractName + 'Proxy') as Truffle.Contract<ProxyInstance>,
-    ])
+    ]
+  })
+}
+
+function fill(a: string) {
+  let res = '__' + a
+  while (res.length < 40) {
```

Comment:
Looks like there's might be a `.padEnd` function? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padEnd

Reply:
Thanks for looking it up!